tests/unit/limitations_test.c

```diff
@@ -245,6 +245,78 @@ CTEST2(limitations, test_io_init_invalid_extent_size)
    }
 }
 
+/*
+ * Test creating SplinterDB with an invalid task system configuration.
+ */
+CTEST2(limitations, test_splinterdb_create_invalid_task_system_config)
+{
+   splinterdb       *kvsb;
+   splinterdb_config cfg;
+   data_config       default_data_cfg;
+
+   default_data_config_init(TEST_MAX_KEY_SIZE, &default_data_cfg);
+   create_default_cfg(&cfg, &default_data_cfg);
+
+   // Both have to be 0, or both have to be set.
+   cfg.num_normal_bg_threads   = 0;
+   cfg.num_memtable_bg_threads = 1;
+
+   int rc = splinterdb_create(&cfg, &kvsb);
+   ASSERT_NOT_EQUAL(0, rc);
+
+   // Cannot use up all possible threads for just bg-threads.
+   cfg.num_normal_bg_threads   = (MAX_THREADS - 1);
+   cfg.num_memtable_bg_threads = 1;
+
+   rc = splinterdb_create(&cfg, &kvsb);
+   ASSERT_NOT_EQUAL(0, rc);
+}
+
+/*
+ * Test that SplinterDB tasks system requires non-zero config for both types
+ * of task groups for background threads to be created.
+ */
+CTEST2(limitations, test_splinterdb_created_w_background_threads)
+{
+   splinterdb       *kvsb;
+   splinterdb_config cfg;
+   data_config       default_data_cfg;
+
+   default_data_config_init(TEST_MAX_KEY_SIZE, &default_data_cfg);
+   create_default_cfg(&cfg, &default_data_cfg);
+
+   // Task system should be setup with background threads
+   cfg.num_normal_bg_threads   = 1;
+   cfg.num_memtable_bg_threads = 1;
+
+   int rc = splinterdb_create(&cfg, &kvsb);
+   ASSERT_EQUAL(0, rc);
+
+   splinterdb_close(&kvsb);
+}
```

Comment:
This test case is a bit odd to me.
The rest of the file is testing "limitations" (almost all being error conditions).
The comment here says it "requires non-zero config for both types of task groups" but that sounds pretty low-level.  It seems like it's just proving that you can initialize a splinterdb with background threads.  And I sort of wonder whether this test maybe instead belongs in `splinterdb_quick_test.c` or `splinterdb_stress_test.c` as a simple "smoke" test that does this setup w/ background threads, and then does a few (300,000?) inserts to trigger at least a memtable incorporation, and then quits?   That might be a nice small demo of the happy-path for this new public API surface.

Reply:
You have a fair point that we can't actually verify that any background work is happening here.

Perhaps the issue for me here is that this "limitations test" file is awkward because it's mixing concerns and creating ambiguity about where test cases ought to live.  Limitations test is trying to focus on error-cases, mostly, but it's doing so for several different components, including both low-level things (task system, io system) and for high-level API (splinterdb.h). So multiple components are mixed in here.  And tests for a single component now span multiple unit-test files.  Further, the presence of an error-case here in limitations is then being used to justify putting an "opposite" happy-path case right below it, also in "limitations test".  So there's a question of whether a happy-path case belongs in this file or in the file that holds all the other happy-path cases for that API.

This doesn't need to be fixed in this PR, but I'm starting to think that this "limitations" file ought to be broken up, and the test cases in it should be moved to the component-specific unit test files where their happy-path cases already live.  That way, all the tests for a component live together in one place, regardless of whether it's covering a happy path or an error path.

This doesn't need to be fixed in this PR.  But I think we ought to move in that direction in the future, perhaps next time we find ourself discussing whether a test-case belongs in `limitations_test.c` or elsewhere.

Reply:
@rosenhouse - noted your concerns about +ve / -ve tests being mixed-up in this limitations_test file. 

For what it's worth, I moved these two cases to `task_system_test.c`:

- test_splinterdb_create_w_background_threads()
- test_splinterdb_create_w_all_background_threads()

Hope this is acceptable to you.